servlet/src/main/java/io/undertow/servlet/api/DeploymentInfo.java

```diff
@@ -315,9 +316,19 @@ public String getDefaultEncoding() {
      */
     public DeploymentInfo setDefaultEncoding(String defaultEncoding) {
         this.defaultEncoding = defaultEncoding;
+        this.defaultEncodingExplicitlySet = true;
         return this;
     }
 
+    /**
+     * Determine whether default encoding was explicitly set
+     *
+     * @return true if default encoding was explicitly set
+     */
+    public boolean isDefaultEncodingExplicitlySet() {
+        return defaultEncodingExplicitlySet;
```

Comment:
I did this to keep backward compatibility. Now defaultEncoding in DeploymentInfo is nullable and check for explicit defaultEncoding set implemented through comparison with null